### PR TITLE
feat (bottom nav review button): v4-830 improve bottom nav review button

### PR DESCRIPTION
**Pull request by mucc001**
Create a new pop-up section similar to the desktop sidebar for the bottom nav review button in the mobile version.

### CODE REVIEW

File: apps/survey/src/components/layouts/meal-list/meal-item.vue [hunk 1]
```diff
@@ -19,12 +19,14 @@
         </v-tooltip>
       </v-list-item-action>
       <v-list-item-action class="my-auto">
-        <context-menu
-          :entity="meal"
-          :entity-name="mealName"
-          v-bind="{ menu }"
-          @action="action"
-        ></context-menu>
+        <v-expand-transition>
```
Comment (lukashroch) [inline apps/survey/src/components/layouts/meal-list/meal-item.vue:22]: not sure if needed yet as like this it won't do much yet

File: apps/survey/src/components/prompts/layouts/base-layout.vue [hunk 2]
```diff
@@ -124,12 +118,16 @@ import type { PropType } from 'vue';
 import { defineComponent } from 'vue';
 
 import type { EncodedFood, MissingFood } from '@intake24/common/types';
+import MealListMobile from '@intake24/survey/components/layouts/meal-list/list-mobile.vue';
+import ReviewList from '@intake24/survey/components/layouts/meal-list/review-list.vue';
```
Comment (lukashroch) [inline apps/survey/src/components/prompts/layouts/base-layout.vue:122]: I think unused import, can be removed

You can also use barrel import
```
import { MealListMobile } from '@intake24/survey/components/layouts/meal-list';
```

File: apps/survey/src/components/prompts/layouts/card-layout.vue [hunk 3]
```diff
@@ -127,11 +121,16 @@
 <script lang="ts">
 import { defineComponent } from 'vue';
 
+import MealListMobile from '@intake24/survey/components/layouts/meal-list/list-mobile.vue';
+import ReviewList from '@intake24/survey/components/layouts/meal-list/review-list.vue';
```
Comment (lukashroch) [inline apps/survey/src/components/prompts/layouts/card-layout.vue:125]: ditto

File: apps/survey/src/components/layouts/meal-list/list-mobile.vue [hunk 4]
```diff
@@ -47,34 +61,70 @@ export default defineComponent({
...
   emits: ['action'],
 
-  setup() {
-    const { getLocaleContent } = useLocale();
-    const { getMealTime } = useMealUtils();
-
-    return { getLocaleContent, getMealTime };
-  },
-
   computed: {
```
Comment (lukashroch) [inline apps/survey/src/components/layouts/meal-list/list-mobile.vue:64]: Can you extract shared logic of desktop and mobile version to composable and reuse for both?

**Review by lukashroch — CHANGES_REQUESTED** [review]

File: apps/survey/src/components/layouts/meal-list/meal-item.vue (hunk 1 above)
Comment (mucc001) [reply apps/survey/src/components/layouts/meal-list/meal-item.vue:22]: Thanks. This doesn't work and I will delete them.

File: apps/survey/src/components/prompts/layouts/base-layout.vue (hunk 2 above)
Comment (mucc001) [reply apps/survey/src/components/prompts/layouts/base-layout.vue:122]: I will remove all the unused imports and use barrel import.

**Pull request merged**